### PR TITLE
Feature/new tables

### PR DESCRIPTION
**Pull request by drreynolds**
Steven and I added some additional embedded explicit Runge--Kutta methods to ARKODE.

### CODE REVIEW

**Review by gardner48 — CHANGES_REQUESTED** [review]

**Comment by gardner48** [comment]
The single precision builds are currently failing with the errors below. I assume the actual table values are within the single precision range and just the rational expressions have constants that are not. Should we note this in the documentation? I'll update some of the CI testing flags shortly.
```
/var/lib/jenkins/workspace/SUNDIALS_GitHub_Test_PR-359/test/tarballs/sundials-6.6.1/src/arkode/arkode_butcher_erk.def:673:1: error: floating constant exceeds range of 'float' [-Werror=overflow]
 ARK_BUTCHER_TABLE(ARKODE_VERNER_9_5_6, { /* Verner-6-5b (DOI: 10.1007/s11075-009-9290-3) */
 ^
/var/lib/jenkins/workspace/SUNDIALS_GitHub_Test_PR-359/test/tarballs/sundials-6.6.1/src/arkode/arkode_butcher_erk.def:733:1: error: floating constant exceeds range of 'float' [-Werror=overflow]
 ARK_BUTCHER_TABLE(ARKODE_VERNER_10_6_7, { /* Verner-7-6 (DOI: 10.1007/s11075-009-9290-3) */
 ^
/var/lib/jenkins/workspace/SUNDIALS_GitHub_Test_PR-359/test/tarballs/sundials-6.6.1/src/arkode/arkode_butcher_erk.def:802:1: error: floating constant exceeds range of 'float' [-Werror=overflow]
 ARK_BUTCHER_TABLE(ARKODE_VERNER_13_7_8, { /* Verner-8-7 (DOI: 10.1007/s11075-009-9290-3) */
```

File: doc/arkode/guide/source/Butcher.rst [hunk 1]
```diff
@@ -790,6 +843,107 @@ This is the default 6th order explicit method (from :cite:p:`Ver:78`).
...
+   The Butcher table is too large to fit in the PDF version of this documentation.  Please see the HTML documentation for the table coefficients.
+
+
+
+.. figure:: /figs/arkode/v65b_erk_stab_region.png
+   :scale: 75 %
+   :align: center
+
+   Linear stability region for the Verner-9-5-6 method.  The method's
+   region is outlined in red; the embedding's region is in blue dashed.
```
[the remark this answers is not in the document]
Comment (drreynolds) [reply doc/arkode/guide/source/Butcher.rst:893]: The current stability region plotting script in our `analysis_utilities` repository colors them this way.  I can update the colors in that script and regenerate the plots, if desired.

**Comment by drreynolds** [comment]
> The single precision builds are currently failing with the errors below. I assume the actual table values are within the single precision range and just the rational expressions have constants that are not. Should we note this in the documentation? I'll update some of the CI testing flags shortly.
> 
> ```
> /var/lib/jenkins/workspace/SUNDIALS_GitHub_Test_PR-359/test/tarballs/sundials-6.6.1/src/arkode/arkode_butcher_erk.def:673:1: error: floating constant exceeds range of 'float' [-Werror=overflow]
>  ARK_BUTCHER_TABLE(ARKODE_VERNER_9_5_6, { /* Verner-6-5b (DOI: 10.1007/s11075-009-9290-3) */
>  ^
> /var/lib/jenkins/workspace/SUNDIALS_GitHub_Test_PR-359/test/tarballs/sundials-6.6.1/src/arkode/arkode_butcher_erk.def:733:1: error: floating constant exceeds range of 'float' [-Werror=overflow]
>  ARK_BUTCHER_TABLE(ARKODE_VERNER_10_6_7, { /* Verner-7-6 (DOI: 10.1007/s11075-009-9290-3) */
>  ^
> /var/lib/jenkins/workspace/SUNDIALS_GitHub_Test_PR-359/test/tarballs/sundials-6.6.1/src/arkode/arkode_butcher_erk.def:802:1: error: floating constant exceeds range of 'float' [-Werror=overflow]
>  ARK_BUTCHER_TABLE(ARKODE_VERNER_13_7_8, { /* Verner-8-7 (DOI: 10.1007/s11075-009-9290-3) */
> ```

I don't see why I can't comment in-line to this item, so I'm answering in a separate comment.

Yes, I think that the issue is that the coefficients are provided in rational form, with components that have a large number of digits.  I could try to convert these to 40-digit decimals to inhibit these warnings, but I'm uncertain how to do this in an automated way (i.e., without manually entering these into Matlab/Mathematica, converting, and pasting the result.

**Review by balos1 — APPROVED** [review]

**Review by gardner48 — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,7 +32,11 @@ Added Fortran support for the LAPACK dense `SUNLinearSolver` implementations.
 Fixed the build system support for MAGMA when using a NVIDIA HPC SDK installation of CUDA
 and fixed the targets used for rocBLAS and rocSPARSE.
 
-Added the fourth order ERK method `ARKODE_SOFRONIOU_SPALETTA_5_3_4`.
+Added the third order ERK method `ARKODE_SHU_OSHER_3_2_3` the fourth order
```
Comment (Steven-Roberts) [inline CHANGELOG.md:35]:
```suggestion
Added the third order ERK method `ARKODE_SHU_OSHER_3_2_3`, the fourth order
```

**Review by gardner48 — APPROVED** [review]

**Pull request merged**